### PR TITLE
Project Private Events

### PR DESCRIPTION
**Pull request by marijanbrvar**
In this project, our main goal is to practice and use data association.
-  Setting up a simple user model
- Adding sign up action to register user no password is required
- Setting up Event model
- Adding association between the user and event creator (user)
- Setting up the Attendee model with an association between the event attendee (also a User) and the event.
- Modify the Event Index page to list all events and adding upcoming and past scopes

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  def new
+    if session[:current_user]
+      @event = Event.new
+    else
+      redirect_to sign_in_path
+    end
+  end
+
+  def create
```
Comment (GzimAsani) [inline app/controllers/events_controller.rb:23]: - [x] uses #build method on association, i.e. `current_user.events.build(event_params)` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/build_method_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+class EventsController < ApplicationController
+  include EventsHelper
+  include UsersHelper
+
+  def index
```
Comment (GzimAsani) [inline app/controllers/events_controller.rb:5]: * [x] lists past events, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists ucomping events, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    if @user.save
+      current_user(@user.id)
+      redirect_to @user
+    else
+      render :new
+    end
+  end
+
+  def show
+    if session[:current_user]
```
Comment (GzimAsani) [inline app/controllers/users_controller.rb:23]: * [x] lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

File: README.md [hunk 4]
```diff
@@ -2,44 +2,29 @@
...
 
-### Install
-
-### Usage
-
-### Run tests
-
-### Deployment
+- To use the application, please clone the repository to your local computer `git clone git@github.com:marijanbrvar/private-events.git` then cd to directory `cd private events
+- Use `bundler` to receive gems required for this project
```
Comment (GzimAsani) [inline README.md]: Be more a bit more descriptive `bundle install`

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# STATUS: Changes Required :red_circle: 

## Hello **Team**, you have done great progress so far :wave: 

### To Highlight:
- Nice readme file :clap: 
- Professional commits :+1: 

### To Improve:

- [x] There should be a button in the events show page where the user can decide to attend that event. Make sure to fix the code in order to give the user that opportunity
- [x] You need to refactor your `Events Controllers / Create Action` check the comment below :arrow_down: 
- [x] You need to refactor your `Events Controllers / Index Action` check the comment below :arrow_down: 
- [x] You need to refactor your `Users Controllers / Show Action` check the comment below :arrow_down: 
- [x] Add both unit and integration tests using using rspec
- [x] Make a proper configuration for a user as an Event attendee ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/attendee_assoc_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [x] In your readme file also add one section on how to setup and run tests

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Hello @marijanbrvar and @zilton7, Great job! 👏  Both of you have done well on this project.
✅ Your codebase looks clean, excellent job 👍 
💯 Your ReadMe file is looking professional 👏 

Let's make a few changes, and we will be done.

- [x] Running individual tests like this(the way you specified on your ReadMe file) can be tedious, imaging have lots of models and controller files. Rspec automatically recognizes tests/specs file with the suffix `_spec`, so immediately your enter `rspec` it reads those files to perform the necessary files.  
However, using `rails generate rspec:model user` where `model` is the type and `user` is the type name. You can also use `rails generate rspec:model event` for event model.
You should try using [shoulda-matchers](https://github.com/thoughtbot/shoulda-matchers) for your unit tests, the domain language is expressive, see sample code below.
```ruby
# RSpec
RSpec.describe MenuItem, type: :model do
  describe 'associations' do
    it { should belong_to(:category) }
  end

  describe 'validations' do
    it { should validate_presence_of(:name) }
    it { should validate_uniqueness_of(:name)
  end
end
```

- [x] Use [Capybara](https://github.com/teamcapybara/capybara) for integration tests, here the tests mimc a typical user activity scenario, open's site => sign up => create event ....

- [x] Add `yarn` to your getting started instructions, perfect after the `bundle install`. I couldn't run the project without running `yarn`

### Nice to have.

- [x] There is no need have an attend button beside *past events*

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Review by uryelah — APPROVED** [review]
Hi **Zil Norvilis & Marian Brvar** , I'm Sarah your reviewer for this PR 👩‍🚀 

Great job implementing the requests from the previous reviewer! 

Your project is approved but I mention an optional changes that can improve your project a little so consider working on it before merging 😉 

Keep on coding and Rock on! 🎸 👨‍🎤 

# Approved 🚀 🎉 

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Pull request merged**